frontend2/src/components/team/EligibilitySettings.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from "react";
+import {
+  useCurrentTeam,
+  TeamStateEnum,
+} from "../../contexts/CurrentTeamContext";
+import SectionCard from "../SectionCard";
+import DescriptiveCheckbox, {
+  CheckboxState,
+} from "../elements/DescriptiveCheckbox";
+import { useEpisode, useEpisodeId } from "../../contexts/EpisodeContext";
+import { isPresent } from "../../utils/utilTypes";
+import { updateTeamPartial } from "../../utils/api/team";
+
+export function determineCheckboxState(
+  inDesired: boolean,
+  inActual: boolean,
+): CheckboxState {
+  if (inDesired && inActual) {
+    return CheckboxState.CHECKED;
+  } else if (!inDesired && !inActual) {
+    return CheckboxState.UNCHECKED;
+  } else {
+    return CheckboxState.LOADING;
+  }
+}
+
+// This component should only be used when there is a logged in user with a team.
+const EligibilitySettings: React.FC = () => {
+  const { team, teamState, refreshTeam } = useCurrentTeam();
+  const episode = useEpisode();
+  const { episodeId } = useEpisodeId();
+  // the desired state according to what the user has clicked on the page
+  const [desiredEligibility, setDesiredEligibility] = useState<
+    number[] | undefined
+  >();
+  useEffect(() => {
+    setDesiredEligibility(team?.profile?.eligible_for);
+  }, [team?.profile?.eligible_for]);
+
+  useEffect(() => {
+    // if actual team eligibility hasn't loaded yet or the desired/actual are identical
+    if (
+      (!isPresent(desiredEligibility) ||
+        desiredEligibility === team?.profile?.eligible_for) ??
```

Comment:
This check doesn't work because the two eligibility arrays aren't the same object. TODO: figure out how to compare these better .. or just remove the check for simplicity?

Reply:
Consider `lodash.eq`?

Reply:
[https://www.geeksforgeeks.org/lodash-_-eq-method/](https://www.geeksforgeeks.org/lodash-_-eq-method/)